internal/controller/nimcache_controller.go

```diff
@@ -816,9 +817,14 @@ func constructJob(nimCache *appsv1alpha1.NIMCache) (*batchv1.Job, error) {
 		if err != nil {
 			return nil, err
 		}
-		if selectedProfiles != nil {
-			job.Spec.Template.Spec.Containers[0].Args = []string{"--profiles"}
-			job.Spec.Template.Spec.Containers[0].Args = append(job.Spec.Template.Spec.Containers[0].Args, selectedProfiles...)
+
+		if len(selectedProfiles) > 0 {
+			if utils.ContainsElement(selectedProfiles, "all") {
+				job.Spec.Template.Spec.Containers[0].Args = []string{"--all"}
+			} else {
+				job.Spec.Template.Spec.Containers[0].Args = []string{"--profiles"}
+				job.Spec.Template.Spec.Containers[0].Args = append(job.Spec.Template.Spec.Containers[0].Args, selectedProfiles...)
```

Comment:
should the profiles be comma-separated?

Reply:
no, they are space delimited currently by the way how NIM container is parsing the CLI args. All arguments following `--profiles` are considered as profiles. We would need to update to comma separated if that logic changes.